File: flipt-java/src/main/java/io/flipt/api/flags/models/FlagType.java
package io.flipt.api.flags.models;

public enum FlagType {
  VARIANT("VARIANT_FLAG_TYPE"),
  BOOLEAN("BOOLEAN_FLAG_TYPE");

  private final String value;

  FlagType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  public static FlagType fromValue(String value) {
    for (FlagType type : FlagType.values()) {
      if (type.value.equals(value)) {
        return type;
      }
    }
    throw new IllegalArgumentException("Unknown flag type: " + value);
  }
}

File: flipt-csharp/src/Models/FlagType.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace Flipt.Models;

/// <summary>
/// Represents the type of flag.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum FlagType
{
    [EnumMember(Value = "VARIANT_FLAG_TYPE")]
    VariantFlagType,

    [EnumMember(Value = "BOOLEAN_FLAG_TYPE")]
    BooleanFlagType
}
